api/lib/domain/usecases/accept-last-pix-terms-of-service.js

```diff
@@ -0,0 +1,6 @@
+module.exports = function acceptLastPixTermsOfService({
+  userId,
+  userRepository
+}) {
+  return userRepository.updateLastPixTermsOfServiceAccepted(userId);
```

Comment:
updateLastPixTermsOfServiceAccepted => acceptLastPixTermsOfService

Reply:
Je trouve cette suggestion de nom un peu trop métier pour du vocabulaire de repo.
Je proposerais plutôt `updatePixLastTermsOfServiceValidation`

Rien de bloquant en revanche

Reply:
J'aime bien que les noms de méthodes de repository soient métier. Le repository est ambiguë car il est une frontière : son interface publique (noms de méthodes, arguments, valeurs de retour et erreurs) est côté métier, alors que son implémentation (SQL ou in memory, ORM) est dans l'infra et ne doit pas arriver dans le métier.
Si on avait des agrégats petits et cohérents, on pourrait aussi avoir des repository beaucoup plus simples, avec un get, un find, un upsert d'agrégat complet et cohérent et un delete, mais pour les CGUs en l'état ça ne fonctionnerait pas bien car ils sont inclus dans le user qui n'est pas un agrégat mais un objet composite complexe.

Reply:
Intéressant !

J'ai l'habitude d'attendre d'un nom de méthode de repo qu'il m'indique grossièrement le genre d'interaction qu'il va avoir avec la BDD, mais c'est peut-être pas toujours adapté.

En relisant les différentes propositions je me rends compte qu'il y a de toute façon cette notion métier portée par `accept` dans le premier cas et `Validation` dans le second.
Peut-être qu'avoir un nom qui commence par `update` me rassure 🤔

Reply:
Merci pour vos retours, du coup on garde la proposition acceptPixLastTermsOfService

Reply:
Je rejoins @sroccaserra, c'est littéralement la raison d'être du _repository_ de permettre au code métier d'ignorer "le genre d'interaction qu'il va avoir avec la BDD". En bon hexagonal, l'interface des _adapters_ est définie par le code métier ("j'aurais besoin de quelque chose qui fait ça") et les implémentations concrètes s'y conforment ("moi je sais faire en utilisant un PostgreSQL", "moi je sais faire avec des données en mémoire"…).